Worker queue documentation (#963)
astronomer/docs: astro/resource-reference-aws.md

| Resource                                  | Description                                                                                                                   | Quantity / Default Size |
| ----------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------- | ----------------------- |
| [EKS Cluster](https://aws.amazon.com/eks) | An EKS cluster is required to run the Astro Data Plane, which hosts the resources and data required to execute Airflow tasks. | 1x                      |
| Worker node pools | Node pools of [EC2 instances](https://aws.amazon.com/ec2/instance-types/) run all Airflow workers. The number of nodes in the pool auto-scales based on the demand for workers in your cluster. You can configure multiple worker node pools to run tasks on different instance types| 1x pool of m5.xlarge nodes |

> | Astro node pool | A node pool of [EC2 instances](https://aws.amazon.com/ec2/instance-types/) runs all proprietary Astronomer components. This node pool is fully managed by Astronomer| 1x pool of m5.xlarge nodes |
> | Airflow node pool | An node pool of [EC2 instances](https://aws.amazon.com/ec2/instance-types/) runs all core Airflow components such as the scheduler and webserver. This node pool is fully managed by Astronomer | 1x pool of m5.xlarge nodes |

Reviewer: ```suggestion
| Airflow system node pool | A node pool of [EC2 instances](https://aws.amazon.com/ec2/instance-types/) that runs all core Airflow components, including the scheduler and webserver. This node pool is fully managed by Astronomer | 1x pool of m5.xlarge nodes |
| Astro system node pool | A node pool of [EC2 instances](https://aws.amazon.com/ec2/instance-types/) that runs all other system components required in Astro. This node pool is fully managed by Astronomer| 1x pool of m5.xlarge nodes |
```

Changed up order -- Airflow system second

- db.m5.16xlarge
- db.m5.24xlarge

### Worker node pools

> Node pools are a scalable collection of worker nodes with the same instance type. These nodes are responsible for running the Pods that execute Airflow tasks. If your cluster has a node pool for a specific instance type, you can configure tasks to run on those instance types using [worker queues](configure-deployment-resources.md#worker-queues.md). To make an instance type available in a cluster, contact [Astronomer support](https://support.astronomer.io) with a request to create a new node pool for the specific instance type. Not all instance types are supported in all AWS regions.

Reviewer: ```suggestion
A node pool is a group of nodes within a cluster that all have the same configuration. On Astro, worker nodes are responsible for running the Pods that execute Airflow tasks. Each worker node pool can be configured with a node instance type and a maximum node count. All Astro clusters have one worker node pool by default, but you can configure additional node pools for advanced configurability. 

If your cluster has multiple worker node pools with different worker node instance types, users in your organization can configure tasks to run on those worker types using [worker queues](configure-deployment-resources.md#worker-queues.md). To enable a new worker type for your cluster, contact [Astronomer support](https://support.astronomer.io) with a request to create a new node pool or modify an existing node pool.
```

Astronomer monitors your usage and the number of nodes deployed in your cluster. As your usage of Airflow increases, Astronomer support might contact you and provide recommendations for updating your node pools to optimize your infrastructure spend or increase the efficiency of your tasks.

For information about each instance type, see [Amazon EC2 Instance Types](https://aws.amazon.com/ec2/instance-types/).

> ### Worker node size resource reference

Reviewer: ```suggestion
### Worker node types
```

> Each worker node in a pool runs a single worker Pod. A worker Pod's actual available size is equivalent to the total capacity of the instance type minus Astro’s system overhead.

Reviewer: ```suggestion
Each worker in a worker node pool runs a single worker Pod. A worker Pod's actual available size is equivalent to the total capacity of the instance type minus Astro’s system overhead.
```

The following table lists all available instance types for worker node pools, as well as the Pod size that is supported for each instance type. As the system requirements of Astro change, these values can increase or decrease.

> | Node Instance Type | CPU       | Memory       |

Reviewer: ```suggestion
| Worker Node Type | CPU       | Memory       |
```

| r6i.4xlarge        | 14 CPUs   | 52.5 GiB MEM |
| r6i.8xlarge        | 30 CPUs   | 112.5 GiB MEM|
| r6i.12xlarge       | 46 CPUs   | 172.5 GiB MEM|
| r6i.16xlarge       | 62 CPUs   | 232.5 GiB MEM|
| r6i.24xlarge       | 94 CPUs   | 352.5 Gib MEM|
| r6i.metal          | 94 CPUs   | 352.5 Gib MEM|
| c6i.xlarge         | 2 CPUs    | 7.5 GiB MEM  |
| c61.2xlarge        | 6 CPUs    | 22.5 GiB MEM |
| c6i.4xlarge        | 14 CPUs   | 52.5 GiB MEM |
| c6i.8xlarge        | 30 CPUs   | 112.5 GiB MEM|
| c6i.12xlarge       | 46 CPUs   | 172.5 GiB MEM|
| c6i.16xlarge       | 62 CPUs   | 232.5 GiB MEM|
| c6i.24xlarge       | 94 CPUs   | 352.5 Gib MEM|
| c6i.metal          | 94 CPUs   | 352.5 Gib MEM|
| c6i.xlarge         | 2 CPUs    | 7.5 GiB MEM  |
| t3.xlarge          | 2 CPUs    | 7.5 GiB MEM  |
| t3.2xlarge         | 6 CPUs    | 22.5 GiB MEM |

:::info

The size limits defined here currently also apply to **Scheduler Resources**, which determines the CPU and memory allocated to the Airflow Scheduler(s) of each Deployment.

For more information about the scheduler, see [Configure a Deployment](configure-deployment-resources.md#scheduler-resources).

:::

> :::tip

Author: ```suggestion
:::info
```

- db.m5.16xlarge
- db.m5.24xlarge

### Worker node pools

> Node pools are a scalable collection of worker nodes with the same instance type. These nodes are responsible for running the Pods that execute Airflow tasks. If your cluster has a node pool for a specific instance type, you can configure tasks to run on those instance types using [worker queues](configure-deployment-resources.md#worker-queues.md). To make an instance type available in a cluster, reach out to [Astronomer support](https://support.astronomer.io) with a request to create a new node pool for the specific instance type. Note that not all instance types are supported in all AWS regions.

Reviewer 2: ```suggestion
Node pools are a scalable collection of worker nodes with the same instance type. These nodes are responsible for running the Pods that execute Airflow tasks. If your cluster has a node pool for a specific instance type, you can configure tasks to run on those instance types using [worker queues](configure-deployment-resources.md#worker-queues.md). To make an instance type available in a cluster, contact [Astronomer support](https://support.astronomer.io) with a request to create a new node pool for the specific instance type. Not all instance types are supported in all AWS regions.
```

> Astronomer monitors your usage and number of nodes deployed in your cluster. As your usage of Airflow increases, Astronomer might reach out with recommendations for updating your node pools to optimize your infrastructure spend or increase the efficiency of your tasks.

Reviewer 2: ```suggestion
Astronomer monitors your usage and the number of nodes deployed in your cluster. As your usage of Airflow increases, Astronomer support might contact you and provide recommendations for updating your node pools to optimize your infrastructure spend or increase the efficiency of your tasks.
```

> For detailed information on each instance type, refer to [AWS documentation](https://aws.amazon.com/ec2/instance-types/).

Reviewer 2: ```suggestion
For detailed information on each instance type, see [Amazon EC2 Instance Types](https://aws.amazon.com/ec2/instance-types/).
```

If you need to pass significant data between Airflow tasks, Astronomer recommends using an [XCom backend](https://airflow.apache.org/docs/apache-airflow/stable/concepts/xcoms.html) such as [AWS S3](https://aws.amazon.com/s3/) or [Google Cloud Storage (GCS)](https://cloud.google.com/storage). For more information and best practices, see the Airflow Guide on [Passing Data Between Airflow Tasks](https://www.astronomer.io/guides/airflow-passing-data-between-tasks).

:::

### Maximum node count

> Each Astro cluster has a limit on how many nodes it can run at once. This maximum includes worker nodes as well as system nodes managed by Astronomer.

Reviewer 2: ```suggestion
Each Astro cluster has a limit on how many nodes it can run at a time. This limit includes the worker nodes and system nodes managed by Astronomer.
```

The default maximum node count for all nodes across your cluster is 20. A cluster's node count is most affected by the number of worker Pods that are executing Airflow tasks. See [Worker autoscaling logic](configure-deployment-resources.md#worker-autoscaling-logic).

> If the node count for your cluster reaches the maximum node count, new tasks might not run or get scheduled. Astronomer monitors maximum node count and is responsible for contacting your organization if it is reached. To check your cluster's current node count, contact [Astronomer Support](https://support.astronomer.io).

Reviewer 2: ```suggestion
If the node count for your cluster reaches the maximum node count, new tasks might not run or get scheduled. Astronomer support monitors the maximum node count and will contact your organization if it is reached. To check your cluster's current node count, contact [Astronomer Support](https://support.astronomer.io).
```
